### ISSUE TITLE
snapshot with new nvarchar column fails

### ISSUE THREAD

**Issue by dubravcik** [dbt-msft/dbt-sqlserver#104]
Hi, I have a table with many nvarchar columns with dbt snapshot on it.

If I add new nvarchar column in the base table, dbt snapshot fails with "String or binary data would be truncated in table...".
Problem is that new column in target snapshot table has length = 1, which is incorrect.

I think the problem is here. The new nvarchar column doesn't have length specified
``` alter table "stg"."salesforce"."account_history" add "domain_c" nvarchar;```

I tried to fix it in catalog macro, but I wasn't successful.

Btw I already added this override so that column adding works (original macro has "alter table ...add column ..." which fails on SQL Server )
``` 
{% macro sqlserver__create_columns(relation, columns) %}
  {% for column in columns %}
    {% call statement() %}
      alter table {{ relation }} add "{{ column.name }}" {{ column.data_type }};
    {% endcall %}
  {% endfor %}
{% endmacro %}
```

**Comment by mikaelene** [comment]
I didn't know adding new columns to a base table and expect to see them in the snapshot-table was supported? I think if you alter the column in the snapshot_destination_table and then try again it will work. 

Just saw below in the docs but have never tried it myself.

We should probably add a test for it in https://github.com/fishtown-analytics/dbt-adapter-tests and then try to fix this with the macro you provided. Thanks!


> When the columns of your source query changes, dbt will attempt to reconcile this change in the destination snapshot table. dbt does this by:

> Creating new columns from the source query in the destination table
Expanding the size of string types where necessary (eg. varchars on Redshift)
dbt will not delete columns in the destination snapshot table if they are removed from the source query. It will also not change the type of a column beyond expanding the size of varchar columns. That is, if a string column is changed to a date column in the snapshot source query, dbt will not attempt to change the type of the column in the destination table.

**Comment by dubravcik** [comment]
Yes, if I alter the column manually in snapshot_destination, it succeeds.

**Comment by swanderz** [comment]
consolidating to #186. @dubravcik #188 fixes your issue officially, and it will be released as part of dbt-sqlserver 0.21.1!
